server/src/main/java/org/apache/iotdb/db/monitor/StatMonitor.java

```diff
@@ -16,317 +16,291 @@
   }
 
-  public long getNumInsert() {
-    return numInsert.get();
-  }
+  /**
+   * Register monitor storage group into system.
+   */
+  public void registerStatGlobalInfo() {
+    PartialPath storageGroupPrefix = new PartialPath(MonitorConstants.STAT_STORAGE_GROUP_ARRAY);
+    try {
+      if (!mManager.isPathExist(storageGroupPrefix)) {
+        mManager.setStorageGroup(storageGroupPrefix);
+      }
 
-  public long getNumInsertError() {
-    return numInsertError.get();
-  }
+      for (StatMeasurementConstants statConstant : StatMeasurementConstants.values()) {
+        PartialPath fullPath = new PartialPath(MonitorConstants.STAT_GLOBAL_ARRAY)
+            .concatNode(statConstant.getMeasurement());
+        registSeriesToMManager(fullPath);
 
-  void registerStatStorageGroup() {
-    MManager mManager = IoTDB.metaManager;
-    PartialPath prefix = new PartialPath(MonitorConstants.getStatStorageGroupPrefixArray());
-    try {
-      if (!mManager.isPathExist(prefix)) {
-        mManager.setStorageGroup(prefix);
+        List<PartialPath> seriesList = monitorSeriesMap
+            .computeIfAbsent(MonitorConstants.STAT_STORAGE_GROUP_NAME, k -> new ArrayList<>());
+        seriesList.add(fullPath);
+        cachedValueMap.putIfAbsent(fullPath, (long) 0);
       }
-    } catch (Exception e) {
-      logger.error("MManager cannot set storage group to MTree.", e);
+    } catch (MetadataException e) {
+      logger.error("Initialize the metadata error.", e);
     }
   }
 
   /**
-   * register monitor statistics time series metadata into MManager.
-   *
-   * @param hashMap series path and data type pair, for example: [root.stat.file.size.DATA, INT64]
+   * Register monitor time series metadata of each storageGroup into MManager.
    */
-  public synchronized void registerStatStorageGroup(Map<String, String> hashMap) {
-    MManager mManager = IoTDB.metaManager;
+  public void registerStatStorageGroupInfo(List<PartialPath> storageGroupNames) {
+    if (storageGroupNames.isEmpty()) {
+      return;
+    }
     try {
-      for (Map.Entry<String, String> entry : hashMap.entrySet()) {
-        if (entry.getValue() == null) {
-          logger.error("Registering metadata but data type of {} is null", entry.getKey());
-        }
-
-        if (!mManager.isPathExist(new PartialPath(entry.getKey()))) {
-          mManager.createTimeseries(new PartialPath(entry.getKey()), TSDataType.valueOf(entry.getValue()),
-              TSEncoding.valueOf("RLE"),
-              TSFileDescriptor.getInstance().getConfig().getCompressor(),
-              Collections.emptyMap());
+      for (PartialPath storageGroupName : storageGroupNames) {
+        if (!storageGroupName.equals(MonitorConstants.STAT_STORAGE_GROUP_NAME)) {
+          // for storage group which is not global, only TOTAL_POINTS is registered now
+          PartialPath fullPath = new PartialPath(MonitorConstants.STAT_STORAGE_GROUP_ARRAY)
+              .concatNode("\"" + storageGroupName + "\"")
```

Comment:
1. storageGroupName contains the seperator ".", we add `"` to see it as the whole.
2. global is surrounded by `"` too to keep consistence, and otherwise `select * from root.stats.global` will throw an exception as global is recognized as a keyword.